lock-api/src/main/java/com/palantir/lock/client/timestampleases/TimestampLeaseAcquirerImpl.java

```diff
@@ -158,22 +174,28 @@ private boolean wasRequestFullyFulfilled(RequestAndResponse requestAndResponse)
                     SafeArg.of("response", requestAndResponse.response()));
         }
 
-        boolean wasFullyFulfilled = requests.keySet().stream().allMatch(timestampName -> {
-            int requestedTimestamps = requests.get(timestampName);
-            long returnedTimestamps =
-                    responses.get(timestampName).getFreshTimestamps().getCount();
-            return returnedTimestamps >= requestedTimestamps;
-        });
+        long wereNotFullyFulfilled = requests.keySet().stream()
+                .filter(timestampName -> {
+                    int requestedTimestamps = requests.get(timestampName);
+                    long returnedTimestamps = responses.get(timestampName).getFreshTimestamps().getCount();
+                    if (returnedTimestamps >= requestedTimestamps) {
+                        return false;
+                    }
+                    telemetryTimestampShortFallConsumer.accept(requestedTimestamps - returnedTimestamps);
+                    return true;
+                })
+                .count();
```

Comment:
This code path will be moved and factored as part of the previous PR reviews, so just review the logic here.

I'll move things later as appropriate